### PR TITLE
Normalization in UEA dataset loading

### PR DESCRIPTION
**Pull request by MrPr3ntice**
Since we do not know whether training and test sets are truly representative samples from the whole population (hard to prove), it is IMHO common best practice to use the parameters (sample mean and -variance) of the training data only for normalization purpose, as long as there is no extra data set left for a final validation. Especially for small test sets, there could hypothetical be an irregular benefit for any later classifier otherwise (Although I don't think that there will be any recognizable change in the metrics here in case of UEA).

see also: https://stats.stackexchange.com/questions/77350/perform-feature-normalization-before-or-within-model-validation

Seems to be the same in case of ucr.py...

### CODE REVIEW

**Comment by White-Link** [comment]
Hi!

This is indeed a relevant question, thanks for raising it. Normalizing should indeed have been done solely using the extracted mean and variance from the training set.

Most of the UCR and UEA datasets are already normalized, so the consequences of this normalization should be limited to few small datasets. The extent of these consequences are unfortunately unclear to me, even though I tend to think that this should not significantly alter the reported results. I believe that F. Liljefors et al. (https://openreview.net/forum?id=HyxQr65z6S), who reproduced our model, did not normalize with the testing set and still obtained similar results to ours.

Regarding any code modifications, I would prefer to keep it in this state for reproducibility purposes with respect to the results reported in our article. However, it may be judicious to highlight it somewhere in the code. I should take care of that in the following days.

Feel free to open an issue on this matter if you want to discuss it further, we are welcoming any feedback!

**Comment by MrPr3ntice** [comment]
Hi,
thanks for your fast reply. I agree with you, that there should not be any recognizable differences in the metrics (As soon I have the time, I will try to run your code on UEA with changed normalization and maybe changed regularization (as one can raise the same issue for early stopping(?)). I will report you, if I see any statistically significant differences to the test performance in your paper). 
I also fully understand the idea of not changing this in the code for reproducibility purposes (Maybe you could add a comment in the code about that for later readers).

Thanks for your work (which I highly appriciate) and also the Link to the NeurIPS reproduction paper - I will read it asap.

Best regards
Philipp

**Comment by White-Link** [comment]
Thanks for your interest in our work! We would be glad to hear about your experiments.
I just added a note about this issue in the comments.
